cellpack/autopack/ingredient/multi_cylinder.py

```diff
@@ -359,3 +386,145 @@ def add_rb_node(self, worldNP):
             )  # math.sqrt(s), 1)# { XUp = 0, YUp = 1, ZUp = 2 } or LVector3f const half_extents
             inodenp.node().addShape(shape, TransformState.makeMat(pMat))  #
         return inodenp
+
+    def checkCylCollisions(
```

Comment:
this looks like a duplication of `collision_jitter`. The idea of splitting these into classes with their own methods is to reduce the amount of duplicate code. It should just be calling `collision_jitter`

Reply:
there is 3 `collision_jitter ` functions. One for sphere in ingredient.py. One for cylinder in multi_cylinder, and one for Cube in single cube. So if we call collision jitter from the parent class, this will call the sphere test and not the cylinder test.

Reply:
the way I was setting this up if the ingredient is an instanced of the mutil_cylinder class, it should use the `collision_jitter` on its class. They're each supposed to have their own implementation of the same method. The one on `Ingredient.py` could be moved to sphere, so there is no default, either way the methods should get overwritten by the subclass, you know for sure it's not working correctly?

Reply:
Yes there is duplication of code in checkCylCollisions and collision_jitter. I added checkCylCollisions  because it was called in Grow.py, realized after its the same code of collision_jitter. I think I made proper change in the latest push.

Reply:
ok cool, yeah, I just realized going through this that I never refactored the `jitter_place` and collision methods on the grow ingredient to match the refactor I did to the other methods. Overall I've been trying to reduce duplicate code because a lot of the different place methods all had the same setup lines and were doing a lot more than just finding if the point can be packed or not. I think the changes you have are ok, but also, since this is just to get the conversion to work, they're not really necessary  here either.